spiketoolkit/sorters/mountainsort4/mountainsort4.py

```diff
@@ -58,116 +54,46 @@ def set_params(self, **params):
-                                     whiten, clip_size, detect_threshold, detect_interval, noise_overlap_threshold)
-
-    print('Elapsed time: ', time.time() - t_start_proc)
-
-    return sorting
-
-
-def _mountainsort4(
-        recording,  # The recording extractor
-        detect_sign=-1,  # Use -1, 0, or 1, depending on the sign of the spikes in the recording
-        adjacency_radius=-1,  # Use -1 to include all channels in every neighborhood
-        freq_min=300,  # Use None for no bandpass filtering
-        freq_max=6000,
-        whiten=True,  # Whether to do channel whitening as part of preprocessing
-        clip_size=50,
-        detect_threshold=3,
-        detect_interval=10,  # Minimum number of timepoints between events detected on the same channel
-        noise_overlap_threshold=0.15  # Use None for no automated curation
-):
-    try:
-        import ml_ms4alg
-    except ModuleNotFoundError:
-        raise ModuleNotFoundError("\nTo use Mountainsort, install ml_ms4alg: \n\n"
-                                  "\npip install ml_ms4alg\n"
-                                  "\nMore information on Mountainsort at: "
-                                  "\nhttps://github.com/flatironinstitute/mountainsort")
-    # Bandpass filter
-    if freq_min is not None:
-        recording = st.preprocessing.bandpass_filter(recording=recording, freq_min=freq_min, freq_max=freq_max)
-
-    # Whiten
-    if whiten:
-        recording = st.preprocessing.whiten(recording=recording)
-
-    # Check location
-    if 'location' not in recording.getChannelPropertyNames():
-        for i, chan in enumerate(recording.getChannelIds()):
-            recording.setChannelProperty(chan, 'location', [0, i])
-
-    # Sort
-    sorting = ml_ms4alg.mountainsort4(
-        recording=recording,
-        detect_sign=detect_sign,
-        adjacency_radius=adjacency_radius,
-        clip_size=clip_size,
-        detect_threshold=detect_threshold,
-        detect_interval=detect_interval
-    )
-
-    # Curate
-    if noise_overlap_threshold is not None:
-        sorting = ml_ms4alg.mountainsort4_curation(
-            recording=recording,
-            sorting=sorting,
-            noise_overlap_threshold=noise_overlap_threshold
+        # Sort
+        p = self.params
+        sorting = ml_ms4alg.mountainsort4(
+            recording=self.recording,
+            detect_sign=p['detect_sign'],
+            adjacency_radius=p['adjacency_radius'],
+            clip_size=p['clip_size'],
+            detect_threshold=p['detect_threshold'],
+            detect_interval=p['detect_interval']
         )
 
-    return sorting
+        # Curate
+        if p['noise_overlap_threshold'] is not None:
+            sorting = ml_ms4alg.mountainsort4_curation(
+                recording=self.recording,
+                sorting=sorting,
+                noise_overlap_threshold=p['noise_overlap_threshold']
+            )
+        self.sorting = sorting
```

Comment:
There is no more self.sorting because it can give a list sorting when several group.

Maybe you can put this in an internal self._sorting_result = [...] initialized with ```n_group *[None]```

And you would get the index with:
```python
ind = self.recording_list.index(recording)
self._sorting_result[ind] = sorting
```

```python
def _get_one_result(self, recording, output_folder):
    ind = self.recording_list.index(recording)
    sorting = self._sorting_result[ind]
    return sorting
```

Reply:
I implemented this in the last commit but `self.recording_list.index(recording)` doesn't seem to work for objects (even though i printed them and they ar ethe same...)
Any idea on how to find the index alternatively?

Reply:
This is strange.